### DISCUSSION TITLE
NES CRC32 failure HW5 Rev3

### DISCUSSION THREAD

**Discussion by Kmeiners** [sanni/cartreader#786] [Need help]
Hello, I built this project a few weeks ago and everything for the most part has been going very smoothly! I have been able to successfully dump GB, GBA, SNES, and N64 carts with no issues whatsoever. 

NES carts have been another story... If I try any of my NES carts (regardless of how clean they are) they will fail the database lookup and then later fail CRC32 validation after trying to dump them manually. I have visually inspected my soldering and tried to test connections for continuity, but haven't noticed anything obvious. I was hoping someone could offer me some guidance on things to try to resolve this issue. 

Edit: I have resoldered the Famicom and NES connectors in case that is the problem, but the issue still persists. Would there be any headers for me to check on the arduino or on the Main or secondary PCBs that are associated with the NES connector?

Edit 2: Interestingly (and maybe somewhat relatedly) I tried using the genesis port and my Sonic the Hedgehog cart has been consistently coming up as "wonig the ledgehog", that seems like all the wrong characters are +4 their intended char value.

**Comment by nsx0r** [comment]
Hello, do you have an example? A full log would help

**Reply by Kmeiners** [reply]
OSCR HW5 V12.4

[+] NES/Famicom


Searching database
for 2E9491AA or 9192D92D...

CRC not found in database
Using manual selection

Printing PRG at 32768
22 00 03 b3 3a 24 c4 40 00 2a 22 22 33 33 33 33 
33 33 32 24 44 44 44 44 44 44 33 33 33 33 33 33 
32 24 44 44 44 44 44 44 11 13 33 33 33 33 39 14 
44 44 44 44 41 11 11 10 00 00 00 00 22 22 00 00 
00 00 01 11 11 10 33 32 24 00 22 22 00 32 24 44 
01 11 11 10 33 32 24 00 2a 22 00 32 24 44 01 11 
11 10 33 32 24 00 22 42 00 32 24 44 01 11 11 10 
00 02 24 00 23 12 00 32 20 00 01 11 33 33 33 32 
24 00 29 12 00 32 24 44 44 44 33 33 33 32 24 00 
31 11 00 32 24 44 44 44 33 33 33 32 24 00 11 11 
00 32 24 44 44 44 11 10 00 02 24 33 19 11 44 32 
20 00 01 11 11 10 33 32 24 33 11 11 44 32 24 44 
01 11 11 10 3b 32 a4 33 11 11 44 b2 2c 44 01 11 
11 10 33 33 33 33 31 14 44 44 44 44 01 11 11 10 
33 33 33 33 31 14 44 44 44 44 01 11 11 10 00 00 
00 22 20 02 22 00 00 00 01 11 11 12 22 22 22 22 
22 22 22 22 22 22 21 11 19 44 c4 4c 44 c4 4c 33 
b3 3b 33 b3 39 11 11 44 44 44 44 44 44 33 33 33 
33 33 31 11 11 44 44 44 44 44 44 33 33 33 33 33 
31 11 00 00 00 01 11 10 00 00 01 11 10 00 00 00 
22 22 22 22 22 22 22 22 22 22 22 22 22 22 3b 33 
3a 44 c4 4c 44 c4 4c 44 c4 24 c4 44 33 33 32 44 
44 44 44 44 44 44 44 24 44 14 11 10 32 44 44 44 
44 44 44 44 44 44 01 11 11 10 02 00 00 00 00 00 
00 00 00 00 01 11 11 33 33 33 33 33 33 33 33 33 
33 33 33 11 11 33 33 33 33 33 31 14 44 44 44 44 
44 11 11 10 11 11 11 11 11 11 11 11 11 11 04 11 
11 10 00 00 00 00 00 00 00 00 00 01 01 11 11 14 
44 44 44 44 41 13 33 33 33 33 31 11 11 14 4c 44 
c4 4c 41 93 3b 33 b3 33 31 11 11 14 44 44 44 44 
41 13 33 33 33 33 31 11 11 10 11 11 11 11 11 11 
11 11 11 11 01 11 11 10 10 00 00 00 00 00 00 00 

Printing PRG at 57344
a9 1b 4c 92 e1 a5 1f 29 01 d0 03 20 39 e1 60 b5 
02 29 f0 c9 f0 f0 2f a5 1f 29 1f d0 07 b5 02 18 
69 10 95 02 b5 02 29 01 f0 06 a5 1f 29 07 d0 13 
a5 1f 29 08 d0 05 a9 3a 4c 3d e4 a9 3c 99 05 02 
99 01 02 a9 00 60 a5 1f 29 18 c9 08 f0 09 c9 10 
f0 0a a9 3c 4c 3d e4 a9 42 4c 3d e4 a9 ff 60 b5 
01 29 02 f0 03 4c 80 e5 ee 39 04 ad 79 04 f0 03 
4c 28 e1 a9 00 85 85 b9 03 02 8d 7e 04 29 07 c9 
04 90 09 b9 03 02 18 69 08 99 03 02 b9 03 02 29 
f8 99 03 02 18 69 04 85 22 b9 00 02 8d 7a 04 29 
07 f0 0b b9 00 02 18 69 08 29 f8 99 00 02 b9 00 
02 18 69 0c 85 21 20 be d0 f0 03 4c 04 e5 20 2a 
d1 f0 0e e6 85 b9 00 02 18 69 08 99 00 02 4c 04 
e5 20 3e d1 20 56 d1 f0 0c b9 03 02 18 69 08 99 
03 02 4c 04 e5 20 2a d1 f0 17 b9 00 02 18 69 08 
99 00 02 b9 03 02 18 69 08 99 03 02 e6 85 4c 04 
e5 4c 28 e1 a9 24 99 05 02 99 01 02 a9 00 99 02 
02 a9 c0 99 06 02 a5 85 d0 1c 98 48 8a 48 b5 01 
29 80 f0 07 a6 42 a0 08 4c 2f e5 a0 00 a6 2e 20 
4e e6 68 aa 68 a8 b9 00 02 8d 7a 04 8d 7b 04 8d 
7c 04 8d 7d 04 99 04 02 b9 03 02 8d 7e 04 8d 7f 
04 8d 80 04 8d 81 04 18 69 08 99 07 02 a9 00 95 
02 20 63 e1 d0 05 a9 43 4c 6d e5 a9 c3 95 01 ee 
79 04 98 48 8a 48 a9 0b 20 81 fa 68 aa 68 a8 60 
ad 7a 04 29 07 d0 39 ad 7a 04 18 69 10 85 21 ad 
7e 04 29 07 d0 3a ad 7e 04 18 69 08 85 22 20 be 
d0 d0 1d 20 3e d1 20 56 d1 d0 07 b5 01 09 80 4c 
bb e5 20 23 d1 d0 19 b5 01 29 7f 95 01 4c d0 e5 
ad 7a 04 c9 f4 b0 1d 18 69 04 8d 7a 04 4c e4 e5 
b5 01 29 80 d0 05 a9 fc 4c dd e5 a9 04 18 6d 7e 
04 8d 7e 04 b5 02 29 03 d0 14 20 3f e7 ad 7d 04 
c9 f4 90 08 a9 00 8d 79 04 4c 28 e1 a9 00 29 03 
Select Mapping from List

NES CART READER

CURRENT SETTINGS

MAPPER:   1
PRG SIZE: 128K
CHR SIZE: 32K
RAM SIZE: 0K

Press Button...

[+] Read iNES Rom


Saving to NES/ROM/BubbleBobble/86/...
[*******************]
CRC32... C7DEC729 -> Not found

Press Button...

**Comment by Kmeiners** [comment]
I think i may have found the problem. I don't suppose the atmega2560 is supposed to be soldered like this? With an unsoldered pad?
![20230509_164209](https://github.com/sanni/cartreader/assets/36490633/b0f37121-6ac1-4d4f-b97c-cbb5ddf31356)

**Reply by Ancyker** [reply]
Yep. The missing pin is digital pin 24. It is used by the NES, Famicom, SNES, and Megadrive/Genesis slots.

![image](https://github.com/sanni/cartreader/assets/7105333/3f6c8612-9b9d-4ed4-aa5c-ecaf638356e4)

**Reply by Kmeiners** [reply]
Well that would certainly explain my issues!

**Reply by Kmeiners** [reply]
This was the problem! Got a new Arduino set up and now all of the problems I was experiencing are fixed. Thanks for your assistance!
